libraries/aws/provisioning/src/aws_iot_provisioning_parser.c

```diff
@@ -53,12 +53,64 @@
  * if successful.
  * @return #AWS_IOT_PROVISIONING_SUCCESS, if parsing is successful; otherwise appropriate error message.
  */
+static AwsIotProvisioningError_t _parseRejectedResponse( IotSerializerDecoderObject_t * pPayloadDecoder,
+                                                         const char * pOperationName,
+                                                         AwsIotProvisioningRejectedResponse_t * pResponseData,
+                                                         AwsIotProvisioningServerStatusCode_t * pStatusCode );
+
+/**
+ * @brief Utility for parsing a text-string based key-value pair entry from
+ * from a map container type response payload.
+ *
+ * @param[in] payloadDecoder The decoder object representing the map container
+ * formatted payload.
+ * @param[in, out] payloadEntryDecoder The decoder object to store the parsed
+ * value data of the entry.
+ * @param[in] pKeyString The key string to parse from the payload.
+ * @param[in] pOperationString The string of the ongoing operation to use for
+ * logging.
+ * @return Returns #AWS_IOT_PROVISIONING_SUCCESS if parsing is successful; otherwise
+ * #AWS_IOT_PROVISIONING_BAD_RESPONSE if any of the expected data entries is missing
+ * in the payload OR @AWS_IOT_PROVISIONING_INTERNAL_FAILURE for decoder failures.
+ */
+static AwsIotProvisioningError_t _parseKeyedEntryInPayload( IotSerializerDecoderObject_t * pPayloadDecoder,
+                                                            IotSerializerDecoderObject_t * pPayloadEntryDecoder,
+                                                            const char * pKeyString,
+                                                            const char * pOperationString );
+
+/**
+ * @brief Common utility for parsing the Certificate PEM string, Certificate ID,
+ * and Ownership token data from the server response.
+ * These elements are common in the server responses of the MQTT CreateKeysAndCertificate
+ * and CreateCertificateFromCsr APIs.
+ * @param[in] payloadDecoder The decoder object that represents the
```

Comment:
Update the variable names in these comments as well. Also, adding a line for spacing before the `@brief`, `@param`, and `@return` sections might look neater.

Reply:
Updated